frontend/taipy-gui/src/components/Taipy/Metric.tsx

```diff
@@ -0,0 +1,113 @@
+import {TaipyBaseProps, TaipyHoverProps} from "./utils";
+import Box from "@mui/material/Box";
+import Skeleton from "@mui/material/Skeleton";
+
+const Plot = lazy(() => import("react-plotly.js"));
+
+interface MetricProps extends TaipyBaseProps, TaipyHoverProps {
+    type?: string
+    min?: number
+    max?: number
+    value?: number
+    defaultValue?: number
+    delta?: number
+    defaultDelta?: number
+    threshold?: number
+    defaultThreshold?: number
+    testId?: string
+    defaultLayout?: string;
+    layout?: string;
+    defaultStyle?: string;
+    style?: string;
+    width?: string | number;
+    height?: string | number;
+}
+
+const emptyLayout = {} as Record<string, Record<string, unknown>>;
+const defaultStyle = {position: "relative", display: "inline-block"};
+
+const Metric = (props: MetricProps) => {
+    const {
+        width = "100%",
+        height
+    } = props;
+    const value = useDynamicProperty(props.value, props.defaultValue, 0)
+    const threshold = useDynamicProperty(props.threshold, props.defaultThreshold, undefined)
+    const delta = useDynamicProperty(props.delta, props.defaultDelta, undefined)
+    const className = useClassNames(props.libClassName, props.dynamicClassName, props.className);
+    const baseLayout = useDynamicJsonProperty(props.layout, props.defaultLayout || "", emptyLayout);
+    const baseStyle = useDynamicJsonProperty(props.style, props.defaultStyle || "", defaultStyle);
+
+    const data = useMemo(() => ([
+        {
+            domain: {x: [0, 1], y: [0, 1]},
+            value: value,
+            type: "indicator",
+            mode: (() => {
+                let mode = "gauge+number";
+                if (delta !== undefined) mode += "+delta";
+                return mode;
+            })(),
+            delta: {
+                reference: typeof value === 'number' && typeof delta === 'number' ? value - delta : undefined,
+            },
+            gauge: {
+                axis: {range: [
+                    typeof props.min === 'number' ? props.min : 0,
+                    typeof props.max === 'number' ? props.max : 100
+                    ]},
+                shape: props.type === "linear" ? "bullet" : "angular",
+                threshold: {
+                    line: {color: "red", width: 4},
+                    thickness: 0.75,
+                    value: threshold
+                }
+            },
+        }
+    ]), [
+        value,
+        delta,
+        threshold,
+        props.min,
+        props.max,
+        props.type,
+    ]);
+
+    const style = useMemo(
+        () =>
+            height === undefined
+                ? ({...baseStyle, width: width} as CSSProperties)
+                : ({...baseStyle, width: width, height: height} as CSSProperties),
+        [baseStyle, height, width]
+    );
+
+    return (
+        <Box data-testid={props.testId} className={className}>
+            <Suspense fallback={<Skeleton key="skeleton"/>}>
```

Comment:
Does Skeleton needs some style (height, width) ?

Reply:
I've added style to the skeleton so it atleast would match with the height & width of the metric